asynch/proto/streams/buffered.py

```diff
@@ -142,11 +142,7 @@ async def read_varint(self):
             packets.append(packet)
             if packet < 0x80:
                 break
-
-        if packets:
-            return leb128.u.decode(packets)
-        else:
-            return None
+        return leb128.u.decode(packets)
```

Comment:
I would really leave this in. Decoding an empty `bytearray` into `0` is clearly wrong behaviour, which caused a very confusing error ("Expected Pong, received Hello packet"). But maybe we should throw an exception instead?

Reply:
I see why it is zero: https://github.com/mohanson/leb128/blob/master/leb128/__init__.py#L27-L33

I prefer raising exception, but which one?

Reply:
Ok, let's the `OperationalError` be chosen.